src/app/info/ClientSideHelpers.tsx

```diff
@@ -1,18 +1,33 @@
 "use client";
 
-import { RoleUpdateContext } from "@/app/roles";
-import { useContext } from "react";
+import { DbWikiRow } from "@/databaseUtils";
+import { useState } from "react";
+import DefaultWikiItemView from "@/app/info/DefaultWikiItemView";
+import EditWikiItemView from "@/app/info/EditWikiItemView";
 
-export const buttonAlert = (): void => {
-    alert("Wiki item has been updated!");
+export const enterEditMode = (setEditMode: (editMode: boolean) => void): void => {
+    setEditMode(true);
 };
 
-interface RoleProps {
-    children?: React.ReactNode;
+interface EditProps {
+    editMode: boolean;
+    row?: DbWikiRow;
 }
 
-export const AdminManagerDependent: React.FC<RoleProps> = (props) => {
-    const { role } = useContext(RoleUpdateContext);
-
-    return <>{(role === "admin" || role === "manager") && props.children}</>;
+export const EditModeDependentItem: React.FC<EditProps> = ({ editMode, row }) => {
+    const [rowRef, setRowRef] = useState<DbWikiRow | undefined>(row);
+    const [editModeRef, setEditModeRef] = useState(editMode);
```

Comment:
as above - could switch to boolean naming convention to make it clearer this is a boolean?